pkg/controller/apmserverelasticsearchassociation/apmserverelasticsearchassociation_controller.go

```diff
@@ -49,6 +51,13 @@ const (
 var (
 	log            = logf.Log.WithName(name)
 	defaultRequeue = reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}
+
+	// apmDefaultRoles are the default roles used by APM Server instances
+	apmDefaultRoles = strings.Join([]string{
+		user.ApmDefaultUserRole, // Retrieve cluster details (e.g. version) and manage apm-* indices
+		"kibana_user",           // Load dependencies, such as example dashboards, if available, into Kibana
+		"ingest_admin",          // Set up index templates
```

Comment:
Can you help me understand the difference between `ApmDefaultUserRole` and this? It seems like we could put `ingest_admin` in the cluster role def there along with `monitor` and `manage_ilm`, right?

Reply:
`apmDefaultRoles` is a collection of roles which contain the role `ApmDefaultUserRole`
`monitor` and `manage_ilm` are privileges, while `ingest_admin` is a built-in role.

I'm not sure we can mix a built-in role along with privileges ?